File: Dante1/Module1.09/Zadanie9.01/main.c
#include <stdio.h>
#include <stdlib.h>
#include "vector_utils.h"

int main()
{
    int tab[150] = { 0 };

    printf("Enter data:\n");

    int size = read_vector(tab,150,0);
    if(size == -2)
    {
        printf("Incorrect input");
        return 1;
    }
    if(size < 1)
    {
        printf("Not enough data available");
        return 3;
    }
    if(size == 1)
    {
        printf("%i",tab[0]);
        return 0;
    }

    int check = bubble_sort(tab, size);
    if(check == 1)
    {
        printf("Incorrect input");
        return 1;
    }
    return 0;
}

File: Dante1/Module1.09/Zadanie9.01/main.rs
mod vector_utils;

fn main()
{
    let mut array:[i32; 150] = [0; 150];

    println!("Enter: ");
    let size: i32 = vector_utils::read_vector(&mut array, 150, 0);
    if size < 1
    {
        panic!("Not enough data available");
    }
    if size == 1
    {
        print!("{}", array[0]);
        return;
    }

    if vector_utils::bubble_sort(&mut array, size as usize) == 1
    {
        panic!("Incorrect input");
    }
}
